WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -1522,6 +1532,25 @@ static GSourceFuncs _handlerIntervention =
             #ifdef WEBKIT_GLIB_API
             WebKitWebContext* context = webkit_web_view_get_context(_view);
             WebKitCookieManager* manager = webkit_web_context_get_cookie_manager(context);
+#if WEBKIT_CHECK_VERSION(2, 42, 0)
+            webkit_cookie_manager_get_all_cookies(manager, NULL, [](GObject* object, GAsyncResult* result, gpointer user_data) {
+                GList* cookies_list = webkit_cookie_manager_get_all_cookies_finish(WEBKIT_COOKIE_MANAGER(object), result, nullptr);
+
```

Comment:
Once again you're completely right 😄 I think I must have tried to do it the wrong way, so assumed I cannot do that here because it is a lambda and ifdef is not "smart" enough. But now when I tried that, it works just fine, so no more need for the code to be duplicated, only function calls based on ifdef.